managed/resource_role.go

```diff
@@ -0,0 +1,379 @@
+/*
+ * Copyright © 2022-present Yugabyte, Inc. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package managed
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	openapiclient "github.com/yugabyte/yugabytedb-managed-go-client-internal"
+)
+
+type resourceRoleType struct{}
+
+func (r resourceRoleType) GetSchema(_ context.Context) (tfsdk.Schema, diag.Diagnostics) {
+	return tfsdk.Schema{
+		Description: `The resource to create a custom role in YugabyteDB Managed.`,
+		Attributes: map[string]tfsdk.Attribute{
+			"account_id": {
+				Description: "The ID of the account this role belongs to.",
+				Type:        types.StringType,
+				Computed:    true,
+			},
+			"project_id": {
+				Description: "The ID of the project this role belongs to.",
+				Type:        types.StringType,
+				Computed:    true,
+			},
+			"role_id": {
+				Description: "The ID of the role. Created automatically when a role is created. Use this ID to get a specific role.",
+				Type:        types.StringType,
+				Computed:    true,
+				Optional:    true,
+			},
+			"role_name": {
+				Description: "The name of the role.",
+				Type:        types.StringType,
+				Required:    true,
+			},
+			"role_description": {
+				Description: "The description of the role.",
+				Type:        types.StringType,
+				Computed:    true,
+				Optional:    true,
+			},
+			"permission_list": {
+				Description: "The permission list of the role.",
+				Required:    true,
+				Attributes: tfsdk.SetNestedAttributes(map[string]tfsdk.Attribute{
+					"resource_type": {
+						Type:     types.StringType,
+						Required: true,
+					},
+					"operation_groups": {
+						Type: types.SetType{
+							ElemType: types.StringType,
+						},
+						Required: true,
+					},
+				}),
+			},
+			"effective_permission_list": {
+				Description: "The effective permission list of the role.",
+				Computed:    true,
+				Attributes: tfsdk.SetNestedAttributes(map[string]tfsdk.Attribute{
+					"resource_type": {
+						Type:     types.StringType,
+						Computed: true,
+					},
+					"operation_groups": {
+						Type: types.SetType{
+							ElemType: types.StringType,
+						},
+						Computed: true,
+					},
+				}),
+			},
+			"users": {
```

Comment:
What is the use case for exposing `users` and `api_keys` to the customer?

Reply:
Similar to how we show the `List of the IDs of the clusters the allow list is assigned to`
But we do not have the ID info of the users or api keys here
If it would be more meaningful to have the IDs of these entities then we would first have to add that info in the apiserver response